draft-reddy-pquip-pqc-hsm.md

```diff
@@ -75,24 +77,20 @@ informative:
 
 --- abstract
 
-
-Hardware Security Modules (HSMs) play a critical role in securing cryptographic operations, including the adoption of Post-Quantum
-Cryptography (PQC). This document examines the use of seed-based key generation in HSMs, which reduces storage requirements but increases
-computational overhead for key derivation. It explores trade-offs between storage efficiency and performance, addressing challenges in
-ephemeral key handling and optimization strategies for PQC signature algorithms. It also discusses PQC impacts to HSM firmware updates and
-backup.
+This document provides guidance for integrating Post-Quantum Cryptography (PQC) into constrained devices, such as IoT devices and lightweight Hardware Security Modules (HSMs), which operate with limited compute, memory, storage, and power. It explores the use of seed-based key generation to minimize persistent storage requirements, strategies for efficient ephemeral key handling, and techniques for offloading cryptographic operations in resource-constrained environments. The document also discusses the impact of PQC on firmware update mechanisms for constrained devices.
 
 --- middle
 
 # Introduction
 
-The transition to post-quantum cryptography (PQC) introduces new challenges for cryptographic key management, particularly within constrained Hardware Security Modules (HSMs). Unlike high-performance, rack-mounted HSMs, constrained HSMs operate with limited memory, storage, and computational resources, making the adoption of PQC algorithms more challenging. The increased key sizes and computational demands of PQC require careful consideration to ensure secure and efficient key management within these constrained environments.
+The transition to post-quantum cryptography introduces significant challenges for resource-constrained devices such as IoT devices, lightweight HSMs, secure elements (e.g., SIMs), and Trusted Execution Environments (TEEs). These devices often operate with limited memory, non-volatile storage, processing power, and battery life, making the adoption of PQC algorithms which typically involve larger key sizes and are more computationally intensive than traditional algorithms particularly challenging. The increased key sizes and computational demands of PQC require careful consideration to ensure secure and efficient key management within these constrained environments.
```

Comment:
Maybe add root of trust to this list?

Reply:
Works for me, please check the below proposed change:

This document provides guidance on integrating Post-Quantum Cryptography (PQC) into resource-constrained devices, including IoT devices and lightweight Hardware Security Modules (HSMs). It highlights the role of the Root of Trust in enabling secure operations, including seed-based key generation to minimize persistent storage, efficient ephemeral key management, offloading cryptographic tasks, and ensuring the integrity of firmware updates and secure boot in PQC-enabled systems.